src/libraries/System.Net.NameResolution/src/System/Net/NameResolutionPal.Windows.cs

```diff
@@ -117,21 +117,25 @@ public static unsafe SocketError TryGetAddrInfo(string name, bool justAddresses,
 
         public static unsafe string GetHostName()
         {
+            // Historically this function has initialized WinSock. We continue to do so.
+            // Trying to debug .NET runtime seeing some ancient vesion of WinSock because a native P/Invoke call
+            // started initializing WinSock first due to rearranged call order is not my idea of a fun day.
+            // I'm not changing it.
             Interop.Winsock.EnsureInitialized();
 
             // We do not cache the result in case the hostname changes.
 
-            const int HostNameBufferLength = 256;
-            byte* buffer = stackalloc byte[HostNameBufferLength];
-            SocketError result = Interop.Winsock.gethostname(buffer, HostNameBufferLength);
-
-            if (result != SocketError.Success)
+            const uint HostNameBufferLength = 256;
+            char* buffer = stackalloc char[(int)HostNameBufferLength];
+            uint bufferused = HostNameBufferLength;
+            if (Interop.Kernel32.GetComputerNameEx(Interop.Kernel32.COMPUTER_NAME_FORMAT.ComputerNameDnsHostname, buffer, ref bufferused) == 0)
```

Comment:
```suggestion
            if (Interop.Kernel32.GetComputerNameEx(Interop.Kernel32.COMPUTER_NAME_FORMAT.ComputerNameDnsHostname, buffer, &bufferused) == BOOL.FALSE)
```

Reply:
(BOOL.FALSE is defined in Interop.BOOL.cs.  If it's not already included in the project, you can add it to the csproj.)